### PR TITLE
OOP School Library: Refactoring the Code

### PR DESCRIPTION
**Pull request by rmiyoyo**
# OOP School Library: Refactoring the Code

### Work Done
- Made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Paid special attention to the "Single-responsibility Principle".
    - The role of `main.rb` is only to start the code.
    - Extracted some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -163,10 +71,10 @@ def choose_option
...
-    when 2 then list_people
-    when 3 then create_person
-    when 4 then create_book
+    when 1 then ListBook.new(@books, self).list_books
+    when 2 then ListPeople.new(@person, self).list_people
+    when 3 then CreatePerson.new(@person, self).create_person
+    when 4 then CreateBook.new(@books, self).create_book
     when 5 then create_rental
     when 6 then list_rental_for_a_person
     when 7 then stop_application
```
Comment (thecodechaser) [inline app.rb]: - [ ] While creating rentals and after entering all the details the app shows `Rental created successfully.` and it quite so users is not able to list the rentals they created. Please make sure to fix this issue and app should not quit after rentals are created and users should be able to list them.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @ismayito, and @rmiyoyo👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with refactoring. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work Done After Review
- We fixed the bug that caused the app to exit after a rental is created.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @ismayito, and @rmiyoyo👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with refactoring. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**